app/scripts/lib/createMetaRPCHandler.js

```diff
@@ -1,6 +1,38 @@
+import log from 'loglevel';
 import { ethErrors, serializeError } from 'eth-rpc-errors';
+import { captureException } from '@sentry/browser';
+import ReadOnlyNetworkStore from './network-store';
+import LocalStore from './local-store';
 
-const createMetaRPCHandler = (api, outStream) => {
+let dataPersistenceFailing = false;
```

Comment:
We're duplicating a lot of things here that are already in `background.js`. Could we move these to a dedicated module instead, and use it in both places? Then we won't be duplicating this logic, and the "data persistence is failing" state.

Reply:
Deduped [here](https://github.com/MetaMask/metamask-extension/pull/15978/commits/f28adb38111b291f75b4f920ff2dd70323a96b9e)